portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +36,39 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final int REQUESTED_COMMENTS_LIMIT = 100;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Sort comments by date from newest to oldest
+    int maxCommentsNumber;
+
+    // Check if max-comments parameter exists
+    if (request.getParameterMap().containsKey("max-comments")) {
+      // Try to convert the value from the 'max-comments' field of the query string to int
+      try {
+        maxCommentsNumber = Integer.parseInt(request.getParameter("max-comments"));
+      } catch (NumberFormatException e) {
+        maxCommentsNumber = REQUESTED_COMMENTS_LIMIT;
+      }
+    } else {
+      response.sendError(400, "'max-comments' parameter is missing.");
+      return;
```

Comment:
The "max-comments" could be an optional parameter. The client should be able to choose to not set it and expect to fetch all the commetns. Or we may want to support pagination. 

When later the API has multiple parameters, we need to decide which parameters are required and which are optional. For optional ones, we need to define the default behavior ideally through API documentation.